### PR TITLE
JBPM-8234 fix failing Marshalling tests

### PR DESCRIPTION
**Pull request by tiagodolphine**
Fix errors on Marshalling tests. 

**Summary:**

TL;DR; 
The problem was at the` BackendDefinitionAdapter.getBaseType`. `UserTask` has changed its parent... now it is `BaseUserTask`, and before it was `BaseTask`, the adapter was just getting the first parent.

Details:
`BootstrapObjectBuilder.stencil` method was getting the Definition based on a given **oryxStencilId**, so sometimes it gets the `UserTask`, and the adapter was not returning the correct BaseType. If the return was any other Task, (BusinessRule, Script, None...) it worked... but this code is totally random. 
If you look at `BaseOryxIdMappings.getKey` this is the random part, it gets the map.entrySet from all the definition mapping and takes the first Definition that matches the **oryxStencilId**, randomly it was getting the UserTask(from the entrySet), and the adapter was not returning the correct BaseType, causing the errors.

Besides that, I closed some `InputStreams` that was not explicity closed on the new Marshhaler, but it seems not to help on the memory leaks during the tests.

@wmedvede 
@romartin 
@xiezhang7 
@hasys 
@manstis

### CODE REVIEW

**Comment by tiagodolphine** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
@tiagodolphine @wmedvede @xiezhang7 @hasys @manstis oh guys great job!! Now it makes sense the random thing... Thanks a lot for your collaboration, has been a pleasure seeing at all the team collaborating to achieve this goal, great job again guys!!! :+1: 

TBH I'm still asking myself WHY CANNOT REMOVE the old marshallers yet???? It's code on Stunner that is actually just causing issues, and we're also not updating it at all, so it's outdated as well. Let's discuss it on Mondaty's call... but that a big yes for me. FYI @krisv @ibek @lazarotti

**Comment by hasys** [comment]
Looks good to me, thank you @tiagodolphine, great catch!

Failed tests in Full downstream build are unrelated.

**Comment by manstis** [comment]
Happy for this to be merged w/o FDB?

**Comment by wmedvede** [comment]
Works wood, @manstis @tiagodolphine  @romartin I think all approvals and builds are in place, So we can go ahead with the merge.
